### PR TITLE
Fixes for Classification ROC

### PR DESCRIPTION
**Pull request by atavory**
Minor fixes for Classification ROC related UDFs

1. Use fall-out for ROC
2. Fixed miss-rate
3. Improved docs

UDFs:

classification_miss_rate
classification_fall_out
classification_precision
classification_recall
classification_thresholds

### CODE REVIEW

**Comment by atavory** [comment]
@tdcmeehan, @highker - opened this instead of a previous diff I rebased incorrectly. Many thanks!

**Comment by tdcmeehan** [comment]
Can you separate this into 3-4 commits?  I don't think the order is important.

1) The bug fix itself
2) The variable renames
3) The new fall out function
4) Tweaks/fixes to the documentation (optional)?

**Comment by atavory** [comment]
@tdcmeehan I'd suggest keeping it as a single (short!) commit, since the things are very much related:

1. There are two fixes: ROC should use fall-out, not miss-rate. This should also be reflected in the docs.
2. The definitions of fall-out and miss-rate were harder to get with the older variables. This is not just a rename - the new variables (whose definitions are slightly different) reflect the exact definitions of the metrics.

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/PrecisionRecallAggregation.java [hunk 1]
```diff
@@ -113,27 +113,30 @@ public static void combine(
 
     protected static class BucketResult
     {
-        public final double totalTrueWeight;
-        public final double totalFalseWeight;
-        public final double runningTrueWeight;
-        public final double runningFalseWeight;
-        public final double left;
-        public final double right;
+        public final double threshold;
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/PrecisionRecallAggregation.java]: Can we create getters for these variables? We don't usually make variables public unless they are static.

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/ClassificationFallOutAggregation.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    private ClassificationFallOutAggregation() {}
+
+    @OutputFunction("array(double)")
+    public static void output(@AggregationState PrecisionRecallState state, BlockBuilder out)
+    {
+        Iterator<BucketResult> resultsIterator = getResultsIterator(state);
+
+        BlockBuilder entryBuilder = out.beginBlockEntry();
+        while (resultsIterator.hasNext()) {
+            final BucketResult result = resultsIterator.next();
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/ClassificationFallOutAggregation.java]: Remove `final`.

Comment (atavory) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/ClassificationFallOutAggregation.java]: Removed it for all the aggregation functions in this PR.

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/ClassificationThresholdsAggregation.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ public static void output(@AggregationState PrecisionRecallState state, BlockBui
             final BucketResult result = resultsIterator.next();
```
[the remark this answers is not in the document]
Comment (atavory) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/ClassificationThresholdsAggregation.java]: Thanks, removed from all similar places.

File: presto-main/src/test/java/com/facebook/presto/operator/TestClassificationMissRateAggregation.java [hunk 4]
```diff
@@ -34,7 +34,7 @@ public TestClassificationMissRateAggregation()
         final ArrayList<Double> expected = new ArrayList<>();
         while (iterator.hasNext()) {
             final BucketResult result = iterator.next();
-            final double missRate = (result.totalFalseWeight - result.remainingFalseWeight) / result.totalTrueWeight;
+            final double missRate = result.remainingFalseWeight / result.totalTrueWeight;
```
Comment (highker) [inline presto-main/src/test/java/com/facebook/presto/operator/TestClassificationMissRateAggregation.java]: Maybe also making `remainingFalseWeight`, etc in `BucketResult` private and provide getters?

**Comment by atavory** [comment]
@highker 

Thanks for your comments! I tried to address them.

File: presto-main/src/test/java/com/facebook/presto/operator/TestClassificationMissRateAggregation.java (hunk 4 above)
Comment (atavory) [reply presto-main/src/test/java/com/facebook/presto/operator/TestClassificationMissRateAggregation.java]: Did that.

**Comment by atavory** [comment]
@wenleix Thanks for your comments! Amended the commit and PR messages according to what you wrote.

**Comment by wenleix** [comment]
Merged #12959 , thanks!

**Pull request merged**